cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -821,6 +830,20 @@ private void emitStartingTimeMetric(ProgramRunId programRunId, long startDelayTi
     metricsContext.gauge(Constants.Metrics.Program.PROGRAM_STARTING_DELAY_SECONDS, startDelayTime);
   }
 
+  private void emitFullyGracefulStopsMetric(ProgramRunId programRunId) {
+    Map<String, String> tags = Collections.emptyMap();
+    MetricsContext metricsContext = ProgramRunners.createProgramMetricsContext(programRunId, tags,
+                                                                               metricsCollectionService);
+    metricsContext.increment(Constants.Metrics.Program.PROGRAM_FULLY_GRACEFUL_STOP_RUNS, 1L);
+  }
+
+  private void emitPartiallyGracefulStopsMetric(ProgramRunId programRunId) {
```

Comment:
Seems like you can combine these two methods into one with a parameter that carries the metrics name?